lib/spack/llnl/util/filesystem.py

```diff
@@ -1389,7 +1389,25 @@ def find_libraries(libraries, root, shared=True, recursive=False):
     # List of libraries we are searching with suffixes
     libraries = ['{0}.{1}'.format(lib, suffix) for lib in libraries]
 
-    return LibraryList(find(root, libraries, recursive))
+    if not recursive:
+        # If not recursive, look for the libraries directly in root
+        return LibraryList(find(root, libraries, False))
+
+    # To speedup the search for external packages configured e.g. in /usr,
+    # perform first non-recursive search in root/lib then in root/lib64 and
+    # finally search all of root recursively. The search stops when the first
+    # match is found.
+    for subdir in ('lib', 'lib64'):
+        dirname = join_path(root, subdir)
+        if not os.path.isdir(dirname):
+            continue
+        found_libs = find(dirname, libraries, False)
```

Comment:
Did you set this false to save time, or because you don't expect this to contain subdirs? This is a slight deviation from the logic transplanted from `spec.py` and may lead to an increase in search time if a library is in a subdirectory

Reply:
I believe it is the same logic as originally in `spec.py` (see https://github.com/spack/spack/pull/11958/files#diff-f8b3ba3a3e9600f6d136b31698f33be7L733).  The logic seems to be:
- check in `root/lib` and `root/lib64` non recursively
- if not found, recursively search `root` (expensive!)